### ISSUE TITLE
Expand SIM118 to cover AnySequence(dict.keys())

### ISSUE THREAD

**Issue by janosh** [charliermarsh/ruff#4262]
Would be nice if SIM118 could be expanded (or a new rule created) that not just covers iterating over dict keys but also converting them to any type of iterable:

```py
# bad
list(dict.keys())
tuple(dict.keys())
set(dict.keys())
[*dict.keys()]
{*dict.keys()}

# good
list(dict)
tuple(dict)
set(dict)
[*dict]
{*dict}
```

**Comment by Skylion007** [comment]
@janosh File this issue as a fix to flake8-comprehensions so that ruff matches the plugin behavior https://github.com/adamchainz/flake8-comprehensions/ :)

**Comment by charliermarsh** [comment]
If flake8-comprehensions implements it, we definitely will, that's an easy call. Otherwise, we have to make the choice to add it to the Ruff rule set. I do think this makes sense to add though.

**Comment by janosh** [comment]
@Skylion007 Do you mean https://github.com/MartinThoma/flake8-simplify?

**Comment by Skylion007** [comment]
Ah yes.

**Comment by charliermarsh** [comment]
We can always add a new rule to `RUF` and redirect it to a `SIM` rule if it gets implemented there.
